[Docs] Specify configs only supported in one architecture

tacole02 opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

#### What this PR does

#### Which issue(s) this PR fixes or relates to

Fixes: 
https://github.com/grafana/mimir-squad/issues/3168 
https://github.com/grafana/mimir-squad/issues/3167
https://github.com/grafana/mimir-squad/issues/3166
https://github.com/grafana/mimir-squad/issues/3148

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`. If changelog entry is not needed, please add the `changelog-not-needed` label to the PR.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

56quarters commented on docs/sources/mimir/configure/about-ingester-circuit-breakers.md:
```diff
@@ -46,6 +46,9 @@ It is possible to configure the maximum allowed duration of both push and read r
 
 A push request meets the push requests circuit breaker failure condition if its duration is longer than the configured maximum push request duration.
 
+{{% admonition type="note" %}}Push requests circuit breakers aren't available in ingest storage architecture. Refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
```
```suggestion
{{% admonition type="note" %}}
Push requests circuit breakers aren't available in ingest storage architecture. Refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
```

56quarters commented on docs/sources/mimir/configure/configure-spread-minimizing-tokens/index.md:
```diff
@@ -9,7 +9,10 @@ title: Migrate ingesters to spread-minimizing tokens
+You can configure Mimir's ingesters to use spread-minimizing token generation.
+
+{{< admonition type="note" >}}
+You only need to configure spread-minimizing token generation if you're running Grafana Mimir with classic architecture. With ingest storage architecture, this process is automatic and doesn't require configuration. Refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).{{< /admonition >}}
```
```suggestion
You only need to configure spread-minimizing token generation if you're running Grafana Mimir with classic architecture. With ingest storage architecture, this process is automatic and doesn't require configuration. Refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
{{< /admonition >}}
```

56quarters commented on docs/sources/mimir/configure/about-reactive-limiters.md:
```diff
@@ -29,6 +29,9 @@ Since inflight limits are, at most, a multiple of the current inflight ingester
 
 Ingesters support separate reactive limiters for push and read requests, since these requests may be overloaded by different resources at different times. When ingesters are heavily loaded, these limiters may be full, and when the load is sustained, queues might fill up as well. When this happens, requests may be rejected, with a preference for rejecting read requests before rejecting push reques ...
 
+{{% admonition type="note" %}}Push request reactive limiters aren't available in ingest storage architecture. Refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
```
```suggestion
{{% admonition type="note" %}}
Push request reactive limiters aren't available in ingest storage architecture. Refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
```

56quarters commented on docs/sources/mimir/configure/configure-zone-aware-replication.md:
```diff
@@ -73,6 +73,9 @@ Deploying Grafana Mimir clusters to fewer zones than the configured replication
 
 A Grafana Mimir deployment should have at least `floor(replication factor / 2) + 1` healthy zones to operate.
 
+{{% admonition type="note" %}}For ingest storage architecture, only two zones are required, regardless of the replication factor. Refer to [About Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
```
```suggestion
{{% admonition type="note" %}}
For ingest storage architecture, only two zones are required, regardless of the replication factor. Refer to [About Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
```

tacole02 commented:
Thanks for reviewing, @56quarters . I fixed the line breaks and removed AGENTS.md. (I had intended to add this to .gitignore, but had inadvertently already push the file).
